lib/cfg-lexer.c

```diff
@@ -450,29 +450,35 @@ cfg_lexer_start_next_include(CfgLexer *self)
       msg_debug("Finishing include",
                 evt_tag_str((level->include_type == CFGI_FILE ? "filename" : "content"), level->name),
                 evt_tag_int("depth", self->include_depth));
-      buffer_processed = TRUE;
+
+      if ((level->include_type == CFGI_BUFFER) ||
+          (level->include_type == CFGI_FILE && !level->file.files))
+        level_finished = TRUE;
+
       cfg_lexer_include_level_close_buffer(self, level);
     }
 
-  if ((level->include_type == CFGI_BUFFER && buffer_processed) ||
-      (level->include_type == CFGI_FILE && !level->file.files))
+  if (level_finished)
```

Comment:
Before this commit, the `if`'s condition was true with `!level->yybuf` and `level->include_type == CFGI_FILE && !level->file.files`, now with `!level->yybuf` we always go to the else branch. Is this intended? Is this a problem?

Reply:
A CfgIncludeLevel struct describes an include operation that can involve including a

* Single file
* buffer
* Or a list of files (e.g. wildcard or directory include).

This function is called at these points:
* When the include operation is not processed yet, we are just going ahead and doing the first include. In this case yybuf is NULL
* When we just finished an include operation and we need to switch to the next file (yybuf non NULL and the files list has items to be processed
* When we finished an include operation and there's nothing more in the files list.

The if branch here decides whether we are finished with the current include level. But this can only happen if yybuf is non NULL, otherwise we are yet to process the very first include.

So I think the old, relatively complex conditions can be simplified, and that's what I did here.

Reply:
Thanks for the explanation :)